### ISSUE TITLE
Incomplete rename because of trailing commas

### ISSUE THREAD

**Issue by susliko** [scalameta/metals#5095]
### Describe the bug

Renaming type `Bar` in the following code
```scala
final case class Foo(
    a: <<Bar>>,
    b: Bar,
)
trait Bar
```
results in `b: Bar` left unrenamed
```scala
final case class Foo(
    a: Quux,
    b: Bar,
)
trait Quux
```

The same rename goes well when there is no trailing comma after `b: Bar`
Tested in VS Code and Nvim

### Expected behavior

All occurences of type `Bar` replaced with `Quux`

### Operating system

macOS

### Editor/Extension

VS Code

### Version of Metals

v0.11.11

**Comment by tgodzik** [comment]
Thanks for reporting! I think it's due to the bug in the compiler that @jkciesluk found, where the position of the last coma is included in the span of the last parameter. We have a special check that makes sure that what gets renamed is actually what we are renaming and this is probably causing the issue.

**Comment by jkciesluk** [comment]
Yes, this is caused by the trailing comma issue. It works fine when the rename is local, i.e.
```scala
def m() = {
  final case class Foo(
      a: Bar,
      b: Bar,
  )
  trait Bar
}
```
But when rename is possibly global (symbol is not defined inside a method), then there is a missmatch (type of B is `Bar,`)

**Comment by susliko** [comment]
Got it
Is there an issue for that compiler bug?

**Comment by jkciesluk** [comment]
Yes, https://github.com/lampepfl/dotty/issues/16872

**Comment by kasiaMarek** [comment]
fixed by: https://github.com/scala/scala3/pull/20445
